SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs

```diff
@@ -12,52 +15,90 @@ public interface IAnonymiseHearingsConferencesDataService
     }
 
-    public class AnonymiseHearingsConferencesDataService : IAnonymiseHearingsConferencesDataService
+    public class
+        AnonymiseHearingsConferencesDataService : IAnonymiseHearingsConferencesDataService
     {
-        private readonly IVideoApiClient _videoApiClient;
         private readonly IBookingsApiClient _bookingsApiClient;
+        private readonly ILogger<AnonymiseHearingsConferencesDataService> _logger;
         private readonly IUserApiClient _userApiClient;
-        public AnonymiseHearingsConferencesDataService(IVideoApiClient videoApiClient, IBookingsApiClient bookingsApiClient,
-            IUserApiClient userApiClient)
+        private readonly IVideoApiClient _videoApiClient;
+        public const string ProcessingUsernameExceptionMessage = "unknown exception when processing {username}";
+
+        public AnonymiseHearingsConferencesDataService(IVideoApiClient videoApiClient,
+            IBookingsApiClient bookingsApiClient,
+            IUserApiClient userApiClient,
+            ILogger<AnonymiseHearingsConferencesDataService> logger)
         {
             _videoApiClient = videoApiClient;
             _bookingsApiClient = bookingsApiClient;
             _userApiClient = userApiClient;
+            _logger = logger;
         }
+
         public async Task AnonymiseHearingsConferencesDataAsync()
         {
-            await DeleteUserAsync();
-            await _videoApiClient.AnonymiseConferencesAsync();
-            await _bookingsApiClient.AnonymiseHearingsAsync();
-        }
+            var anonymisationData = await _bookingsApiClient.GetAnonymisationDataAsync();
+
+            if (anonymisationData.HearingIds != null && anonymisationData.HearingIds.Any())
+            {
+                _logger.LogInformation("Hearing ids being processed: {hearingids}", anonymisationData.HearingIds);
+                
+                await _videoApiClient.AnonymiseConferenceWithHearingIdsAsync(new AnonymiseConferenceWithHearingIdsRequest
+                    { HearingIds = anonymisationData.HearingIds });
+                
+                await _videoApiClient.AnonymiseQuickLinkParticipantWithHearingIdsAsync(
+                    new AnonymiseQuickLinkParticipantWithHearingIdsRequest { HearingIds = anonymisationData.HearingIds });
+
+                await _bookingsApiClient.AnonymiseParticipantAndCaseByHearingIdAsync("hearingIds",
+                    anonymisationData.HearingIds);
+            }
 
-        private async Task DeleteUserAsync()
-        {
-            // delete users from AAD.
-            // get users that do not have hearings in the future and have had hearing more than 3 months in the past. 
-            // (exclude judges, vhos, test users, performance test users.
-            var usersToDelete = await _bookingsApiClient.GetPersonByClosedHearingsAsync();
 
-            if (usersToDelete != null && usersToDelete.Usernames != null)
+            if (anonymisationData.Usernames == null) return;
```

Comment:
Is it worth a log here to indicate there was nothing to process

Reply:
I dont think we need to log this, especially because we are also logging all the hearing ids being processed earlier which should give enough information about what is being processed.